example/lib/src/ble/ble_logger.dart

```diff
@@ -0,0 +1,15 @@
+import 'package:intl/intl.dart';
+
+class BleLogger {
+  final List<String> _logMessages = [];
+  final DateFormat formatter = DateFormat('hh:mm:ss');
+
+  List<String> get messages => _logMessages;
+
+  void addToLog(String message) {
+    final now = DateTime.now();
+    _logMessages.add('${formatter.format(now)}.${now.millisecond} - $message');
```

Comment:
Couldn't you use the date formatter to render fractions of seconds. I believe, it must be something like `hh:mm:ss.SSS`. Try it of check the spec.